crates/bevy_text/src/text.rs

```diff
@@ -23,160 +24,152 @@ impl Default for CosmicBuffer {
+    ///
+    /// The [`TextEntity::depth`] field can be used to reconstruct the hierarchy.
+    pub(crate) entities: SmallVec<[TextEntity; 1]>,
+    /// Flag set when any change has been made to this block that should cause it to be rerendered.
+    ///
+    /// Includes:
+    /// - [`TextBlock`] changes.
+    /// - [`TextStyle`] or `Text2d`/`Text`/`TextSpan2d`/`TextSpan` changes anywhere in the block's entity hierarchy.
+    // TODO: This encompasses both structural changes like font size or justification and non-structural
+    // changes like text color and font smoothing. This field currently causes UI to 'remeasure' text, even if
+    // the actual changes are non-structural and can be handled by only rerendering and not remeasuring. A full
+    // solution would probably require splitting TextBlock and TextStyle into structural/non-structural
+    // components for more granular change detection. A cost/benefit analysis is needed.
+    pub(crate) needs_rerender: bool,
+}
+
+impl ComputedTextBlock {
+    /// Accesses entities in this block.
+    ///
+    /// Can be used to look up [`TextStyle`] components for glyphs in [`TextLayoutInfo`] using the `span_index`
+    /// stored there.
+    pub fn entities(&self) -> &[TextEntity] {
+        &self.entities
+    }
+
+    /// Indicates if the text needs to be refreshed in [`TextLayoutInfo`].
+    ///
+    /// Updated automatically by [`detect_text_needs_rerender`](crate::detect_text_needs_rerender) and cleared
+    /// by [`TextPipeline`] methods.
+    pub fn needs_rerender(&self) -> bool {
+        self.needs_rerender
+    }
+}
+
+impl Default for ComputedTextBlock {
+    fn default() -> Self {
+        Self {
+            buffer: CosmicBuffer::default(),
+            entities: SmallVec::default(),
+            needs_rerender: true,
+        }
+    }
+}
+
+/// Component with text format settings for a block of text.
+///
+/// A block of text is composed of text spans, which each have a separate string value and [`TextStyle`]. Text
+/// spans associated with a text block are collected into [`ComputedTextBlock`] for layout, and then inserted
+/// to [`TextLayoutInfo`] for rendering.
+///
+/// See [`Text2d`] for the core component of 2d text, and `Text` in `bevy_ui` for UI text.
 #[derive(Component, Debug, Clone, Default, Reflect)]
 #[reflect(Component, Default, Debug)]
-pub struct Text {
-    /// The text's sections
-    pub sections: Vec<TextSection>,
+#[require(ComputedTextBlock, TextLayoutInfo)]
+pub struct TextBlock {
     /// The text's internal alignment.
     /// Should not affect its position within a container.
     pub justify: JustifyText,
-    /// How the text should linebreak when running out of the bounds determined by `max_size`
+    /// How the text should linebreak when running out of the bounds determined by `max_size`.
     pub linebreak: LineBreak,
     /// The antialiasing method to use when rendering text.
     pub font_smoothing: FontSmoothing,
```

Comment:
I think don't think the font smoothing setting belongs here, it doesn't affect the arrangement of the text. It would be better stored in an elidable component of its own or in `TextStyle`.

Reply:
Moved to TextStyle since smoothing is applied per-font. Can break it into a standalone component in a separate PR.